ansible_runner/utils/streaming.py

```diff
@@ -0,0 +1,59 @@
+import tempfile
+import zipfile
+import os
+import json
+import sys
+
+from .base64io import Base64IO
+from pathlib import Path
+
+def stream_dir(source_directory, stream):
+    with tempfile.NamedTemporaryFile() as tmp:
+        with zipfile.ZipFile(tmp.name, 'w', compression=zipfile.ZIP_DEFLATED, allowZip64=True) as archive:
+            if source_directory:
+                for dirpath, dirs, files in os.walk(source_directory):
+                    relpath = os.path.relpath(dirpath, source_directory)
+                    if relpath == ".":
+                        relpath = ""
+                    for fname in files:
+                        archive.write(os.path.join(dirpath, fname), arcname=os.path.join(relpath, fname))
+            archive.close()
+
+        zip_size = Path(tmp.name).stat().st_size
+
+        with open(tmp.name, "rb") as source:
+            if stream.name == '<stdout>':
+                target = sys.stdout.buffer
+            else: 
+                target = stream
+            target.write(json.dumps({'zipfile': zip_size}).encode('utf-8') + b'\n')
+            with Base64IO(target) as encoded_target:
+                for line in source:
+                    encoded_target.write(line)
+
+
+def unstream_dir(stream, length, target_directory):            
+    # NOTE: caller needs to process exceptions
+    with tempfile.NamedTemporaryFile() as tmp:
+        with open(tmp.name, "wb") as target:
+            with Base64IO(stream) as source:                
+                while True:
+                    remaining = length
```

Comment:
this is in the loop, so it's basically always going to be getting reset to the full size

Reply:
Nice catch!